core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -158,21 +161,36 @@ private void tryStart() {
             containerIsStarting(containerInfo);
 
             // Wait until the container is running (may not be fully started)
-            profiler.start("Wait until container state=running");
-            Unreliables.retryUntilTrue(30, TimeUnit.SECONDS, () -> {
+            profiler.start("Wait until container state=running, or there's evidence it failed to start.");
+            final Boolean[] startedOK = {null};
+            Unreliables.retryUntilTrue(CONTAINER_RUNNING_TIMEOUT_SEC, TimeUnit.SECONDS, () -> {
                 //noinspection CodeBlock2Expr
                 return DOCKER_CLIENT_RATE_LIMITER.getWhenReady(() -> {
                     InspectContainerResponse inspectionResponse = dockerClient.inspectContainerCmd(containerId).exec();
-                    return inspectionResponse.getState().isRunning();
+                    if (inspectionResponse.getState().isRunning()) {
+                        startedOK[0] = true;
+                        return true;
+                    } else if (inspectionResponse.getState().getFinishedAt() != null) {
+                        // container started, but is now not running. That means it started but has since stopped;
+                        // likely due to misconfiguration.
+                        startedOK[0] = false;
+                        return true;
+                    }
+                    return false;
                 });
             });
 
+            if (!startedOK[0]) {
+                // Bail out, don't wait for the port to start listening.
+                // (Exception thrown here will be caught below and wrapped)
+                throw new NotFoundException("Container has already stopped.");
```

Comment:
I feel slightly funny about using a client lib's exception class - I think I'll amend to a boring-but-neutral `IllegalStateException` given that it will invariably be wrapped anyway.